### DISCUSSION TITLE
VoiP on UDMPro

### DISCUSSION THREAD

**Discussion by iniciante** [fabianishere/udm-iptv#28] [Ideas]
Hi,
In my case there are 3 Vlans (internet(6), voip(3) and Iptv(4)). I was wondering how to enable the container to manage Voip. I come from Erpoe router where the configuration was:
set interfaces ethernet eth9 vif 3 address dhcp
set interfaces ethernet eth9 vif 3 description Voip
set interfaces ethernet eth9 vif 3 dhcp-options default-route no-update
set interfaces ethernet eth9 vif 3 egress-qos '0:4 1:4 2:4 3:4 4:4 5:4 6:4 7:4'
set protocols rip interface eth9.3
set protocols rip passive-interface default
set service nat rule 5001 description Masq_Voip
set service nat rule 5001 log disable
set service nat rule 5001 outbound-interface eth9.3
set service nat rule 5001 protocol all

may I ask for your help reproducing it on the udmpro?

Thanks a lot.

**Comment by fabianishere** [comment]
You could first try setting it up manually:
```
ip link add link eth8 name voip type vlan id 3
udhcpc -d /bin/true -s /bin/true -i voip
iptables -t nat -A POSTROUTING -j MASQUERADE -o voip
```

**Comment by jwillemsen** [comment]
Would it be an option to extend `install.sh` with support for VOIP? I have a UDM Pro SE with KPN and would like to use VOIP together with a Siemens N300A gigaset base station.

**Reply by fabianishere** [reply]
I don't use VOIP, so I have no way of building/testing these changes myself. I am happy to accept contributions for this functionality though.

**Reply by fabianishere** [reply]
By the way, VOIP with KPN runs over the internet VLAN, so it should probably work out of the box.

**Reply by jwillemsen** [reply]
Thanks for the details, will try this asap, first step is to move VOIP from the KPN modem, next step is to use the UDM Pro SE for all traffic.

**Comment by iniciante** [comment]
Maybe I am newbie...
I had previously a vlan named voip on a port of the udmpro, only to be connected by the Voip device. With Vlan 3.

I have run your commands after remove my previous vlan but the result is
# udhcpc -d /bin/true -s /bin/true -i voip
udhcpc: started, v1.31.1
udhcpc: sending discover
udhcpc: sendto: Network is down
udhcpc: read error: Network is down, reopening socket

I am running the commands over the symbol "#", is it correct?
Why eht8? Is it the port on the udmpro where comes in the connection to the internet?
If the commands are successful, how do you connect to that vlan? I mean if you need additionally to create Lan named Vlan with number 3 in my case (Movistar).

Thanks a lot.

**Reply by fabianishere** [reply]
Probably, you need the following as well:
```
ip link set voip up
```

The VLAN device is created using `eth8` as parent, since that is the WAN port on the UDM Pro. On the `voip` interface, the VOIP packets will arrive from the WAN side (VLAN 3). You might still need to setup the correct routes for the VOIP device. However, I don't have any experience with this myself, so this might miss some steps.

**Comment by iniciante** [comment]
Now it has best aspects:

voip      Link encap:Ethernet  HWaddr 74:AC:B9:14:2A:8D  
          inet6 addr: fe80::76ac:b9ff:fe14:2a8d/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:13 errors:0 dropped:0 overruns:0 frame:0
          TX packets:9 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:1326 (1.2 KiB)  TX bytes:1490 (1.4 KiB)

The question now is how to engage the Gigabit Voip device. Should I create a vlan with flag 3 to connect it or it is available for all the connected devices to the router/switch?

Thanks.

**Reply by fabianishere** [reply]
You'll need to setup the appropriate routes, but this depends on your VOIP provider.

**Comment by iniciante** [comment]
The info I have from Edgerouter, where it was working, is:

 ethernet eth9 {
        duplex auto
        speed auto
        vif 3 {
            address dhcp
            description Voip
            dhcp-options {
                default-route no-update
                default-route-distance 210
                name-server update
            }
            egress-qos "0:4 1:4 2:4 3:4 4:4 5:4 6:4 7:4"
        }
Is this what you mean?

**Comment by iniciante** [comment]
Or using Cli it was

configure
set interfaces ethernet eth9 vif 3 address dhcp
set interfaces ethernet eth9 vif 3 description Voip
set interfaces ethernet eth9 vif 3 dhcp-options default-route no-update
set interfaces ethernet eth9 vif 3 egress-qos '0:4 1:4 2:4 3:4 4:4 5:4 6:4 7:4'
set protocols rip interface eth9.3
set protocols rip passive-interface default
set service nat rule 5001 description Masq_Voip
set service nat rule 5001 log disable
set service nat rule 5001 outbound-interface eth9.3
set service nat rule 5001 protocol all
save;exit


I think I should replace eth9 by eth8 and eth9.3 by eth8.3, is it correct do you think?

**Comment by iniciante** [comment]
Not for Movistar, it is a different vlan
